src/PhoneNumberFormatService.php

```diff
@@ -0,0 +1,51 @@
+<?php
+declare(strict_types=1);
+
+namespace DR\Internationalization;
+
+use DR\Internationalization\PhoneNumber\PhoneNumberFormatOptions;
+use InvalidArgumentException;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumber;
+use libphonenumber\PhoneNumberUtil;
+use RuntimeException;
+
+class PhoneNumberFormatService
+{
+    private PhoneNumberFormatOptions $defaultOptions;
+    private ?PhoneNumberUtil $phoneNumberUtil = null;
+
+    public function __construct(PhoneNumberFormatOptions $phoneNumberOptions)
+    {
+        $this->defaultOptions = $phoneNumberOptions;
+    }
+
+    public function format(string $phoneNumber, PhoneNumberFormatOptions $options = null): string
+    {
+        $countryCode = $options?->getDefaultCountryCode() ?? $this->defaultOptions->getDefaultCountryCode();
+        $format      = $options?->getFormat() ?? $this->defaultOptions->getFormat();
+        if ($format === null) {
+            throw new InvalidArgumentException('PhoneNumberOptions: unable to format phoneNumber without a given format');
```

Comment:
And this is more a RuntimeException? Or ConfigurationException

Reply:
Used the same exception as the NumberFormatService here